pypesto/sampling/metropolis.py

```diff
@@ -29,31 +31,41 @@ def default_options(cls):
 
     def initialize(self, problem: Problem, x0: np.ndarray):
         self.problem = problem
-        self.objective = problem.objective
-        self.objective.history = History()
+        self.neglogpost = problem.objective
+        self.neglogpost.history = History()
+        if problem.x_priors is None:
+            self.neglogprior = lambda x: -0.
+        else:
+            self.neglogprior = problem.x_priors
         self.trace_x = [x0]
-        self.trace_fval = [self.objective(x0)]
+        self.trace_neglogpost = [self.neglogpost(x0)]
+        self.trace_neglogprior = [self.neglogprior(x0)]
 
-    def sample(self, n_samples: int, beta: float = 1.):
+    def sample(self, n_samples: int, beta: float = 1.,
+               temper_lpost: bool = False):
         # load last recorded particle
         x = self.trace_x[-1]
-        llh = - self.trace_fval[-1]
+        lpost = - self.trace_neglogpost[-1]
+        lprior = - self.trace_neglogprior[-1]
 
         show_progress = self.options['show_progress']
 
         # loop over iterations
         for _ in tqdm(range(int(n_samples)), disable=not show_progress):
             # perform step
-            x, llh = self._perform_step(x, llh, beta)
-
+            x, lpost, lprior = self._perform_step(x, lpost, lprior,
+                                                  beta, temper_lpost)
```

Comment:
please use kwargs to avoid passing lpost, lprior the wrong way round.

Reply:
I think this is addressed now, let me know whether this suffices